vortex-api/src/main/java/mil/army/usace/hec/vortex/io/NetcdfDataReader.java

```diff
@@ -716,22 +738,34 @@ private VortexData getData(GridDataset dataset, String variable, int idx) {
                 .startTime(startTime)
                 .endTime(endTime)
                 .interval(interval)
+                .dataType(vortexDataType)
+                .vortexVariable(vortexVariable)
                 .build();
     }
 
     private float[] getData(float[] slice, GridCoordSystem gcs, Grid gridDefinition, double noDataValue) {
         if (gcs.isRegularSpatial())
             return slice;
 
+        support.firePropertyChange(VortexProperty.STATUS, null, "ReIndexing");
+
         IndexSearcher indexSearcher = IndexSearcherFactory.INSTANCE.getOrCreate(gcs);
         Coordinate[] coordinates = gridDefinition.getGridCellCentroidCoords();
         float[] data = new float[coordinates.length];
-        for (int i = 0; i < data.length; i++) {
+        int count = data.length;
+        for (int i = 0; i < count; i++) {
             Coordinate coordinate = coordinates[i];
             int index = indexSearcher.getIndex(coordinate.x, coordinate.y);
             data[i] = index >= 0 ? slice[index] : (float) noDataValue;
+            int progress = (int) (((float) i / count) * 100);
+            support.firePropertyChange(VortexProperty.PROGRESS, null, progress);
         }
 
         return data;
     }
+
+    private VortexDataType getVortexDataType(VariableDS variableDS) {
```

Comment:
This function serves to get the `VortexDataType` for the NetCDF file via the `CELL_METHODS` attribute.